### PR TITLE
DWARF register numberings

### PR DESCRIPTION
**Pull request by mshinwell**
This exciting pull request defines the mappings between the compiler's internal hard register numbering schemes and the DWARF register numbering schemes for the various platforms.  It also defines which DWARF register number on each platform corresponds to the system stack pointer.

I am presuming that modern ELF but non-Linux platforms such as the common BSD systems use the same conventions as Linux systems in these regards.

### CODE REVIEW

**Comment by mshinwell** [comment]
@bschommer Could you perhaps check this?

**Comment by bschommer** [comment]
I can take a look at it.

File: asmcomp/power/proc.ml [hunk 1]
```diff
@@ -257,6 +257,43 @@ let loc_external_results res =
...
+*)
+
+let int_dwarf_reg_numbers =
+  [| 3; 4; 5; 6; 7; 8; 9; 10;
+     14; 15; 16; 17; 18; 19; 20; 21;
+     22; 23; 24; 25; 26; 27; 28;
+  |]
+
+let float_dwarf_reg_numbers =
+  [| 32; 33; 34; 35; 36; 37; 38; 39;
```
Comment (bschommer) [inline asmcomp/power/proc.ml]: This should start with 33, since the float register 1 has the mapping 33.

File: asmcomp/arm/proc.ml [hunk 2]
```diff
@@ -224,6 +224,29 @@ let loc_external_results res =
 
 let loc_exn_bucket = phys_reg 0
 
+(* See "DWARF for the ARM architecture" available from developer.arm.com. *)
+
+let int_dwarf_reg_numbers =
+  [| 0; 1; 2; 3; 4; 5; 6; 7; 12 |]
+
+let float_dwarf_reg_numbers =
```
Comment (bschommer) [inline asmcomp/arm/proc.ml:239]: For armv6 the registers numbers are 64-95, since this is actually meant with `Legacy VFP v2-use`.

**Review by bschommer — CHANGES_REQUESTED** [review]
Looks good overall with two minor problems.

File: asmcomp/arm/proc.ml (hunk 2 above)
Comment (mshinwell) [reply asmcomp/arm/proc.ml:239]: I think I've fixed this.

File: asmcomp/power/proc.ml (hunk 1 above)
Comment (mshinwell) [reply asmcomp/power/proc.ml]: Well spotted.

**Comment by mshinwell** [comment]
@bschommer Thanks for the review, perhaps you could check `5a171c4`.

**Comment by bschommer** [comment]
Looks good to me.

**Review by gasche — APPROVED** [review]
Approved on the basis of @bschommer's review. (I haven't reviewed myself.)

**Pull request merged**